plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java

```diff
@@ -0,0 +1,107 @@
+package org.radargun.http.service;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.Response;
+
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.radargun.traits.CacheInformation;
+
+/**
+ * Providing cache info over resteasy-client.
+ * @author Anna Manukyan
+ */
+public class RESTEasyCacheInfo implements CacheInformation {
+   private static final String CONTENT_TYPE = "text/plain";
+   private RESTEasyCacheService service;
+
+   public RESTEasyCacheInfo(RESTEasyCacheService service) {
+      this.service = service;
+   }
+
+   @Override
+   public String getDefaultCacheName() {
+      return service.cacheName;
+   }
+
+   @Override
+   public Collection<String> getCacheNames() {
+      return Arrays.asList(service.cacheName);
+   }
+
+   @Override
+   public Cache getCache(String cacheName) {
+      return new Cache(service.getHttpClient());
+   }
+
+   protected class Cache implements CacheInformation.Cache {
+      protected ResteasyClient httpClient;
+
+      public Cache(ResteasyClient httpClient) {
+         this.httpClient = httpClient;
+      }
+
+      @Override
+      public long getOwnedSize() {
+         return -1;
+      }
+
+      @Override
+      public long getLocallyStoredSize() {
+         return -1;
+      }
+
+      @Override
+      public long getMemoryStoredSize() {
+         return -1;
+      }
+
+      @Override
+      public long getTotalSize() {
+         long size = 0;
+         if (service.isRunning()) {
+            String target = service.buildUrl(null);
+            Response response = null;
+            String responseStr = null;
+            try {
+               Invocation get = service.getHttpClient().target(target).request().accept(CONTENT_TYPE).buildGet();
+               response = get.invoke();
+
+               responseStr = response.readEntity(String.class);
+               size = Long.parseLong(responseStr);    //if the call was done to nodejs app, then the long size is returned
+
+            } catch (NumberFormatException ex) {
+               //If the call was done to Infinispan Server, then the response is the list of all entries
+               String[] entries = responseStr.split(System.getProperty("line.separator"));
```

Comment:
Ah. Actually it returns just keys. Not whole entries. This might be fine if used rarely. What's the purpose of this method? When is it used? How often?